fulp_modules/QM_wrestling/code/wrestling.dm

```diff
@@ -0,0 +1,455 @@
+			to_chat(user, "<span class='warning'>[target] is too far away!</span>")
+			return
+		if(!isturf(user.loc) || !isturf(target.loc))
+			to_chat(user, "<span class='warning'>You can't slam [target] here!</span>")
+			return
+		target.forceMove(user.loc)
+		var/fluff = "body-slam"
+		target.visible_message("<span class='danger'>[user] [fluff] [target]!</span>", \
+						"<span class='userdanger'>You're [fluff]ed by [user]!</span>", "<span class='hear'>You hear a sickening sound of flesh hitting flesh!</span>", COMBAT_MESSAGE_RANGE, user)
+		to_chat(user, "<span class='danger'>You [fluff] [target]!</span>")
+		playsound(user.loc, "swing_hit", 50, TRUE)
+		if(!target.stat)
+			target.emote("scream")
+			target.Paralyze(30)
+			target.adjustBruteLoss(rand(10,15))
+		else
+			target.adjustBruteLoss(rand(10,15))
+	else
+		if(user)
+			user.pixel_x = user.base_pixel_x
+			user.pixel_y = user.base_pixel_y
+		if(target)
+			target.pixel_x = target.base_pixel_x
+			target.pixel_y = target.base_pixel_y
+
+	log_combat(user, target, "body-slammed (Wrestling)")
+	return
+
+/datum/martial_art/fulpwrestling/proc/FlipAnimation(mob/living/D)
+	set waitfor = FALSE
+	if(D)
+		animate(D, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
+	if(do_after(D, 10 SECONDS))
+		if(D)
+			animate(D, transform = null, time = 1, loop = 0)
+
+/// Drop
+/datum/martial_art/fulpwrestling/proc/fulpdrop(mob/living/user, mob/living/target)
+	if(!can_use(user))
+		return FALSE
+	if(!target)
+		return
+	var/obj/surface = null
+	var/turf/ST = null
+	var/falling = 0
+
+	for(var/obj/O in oview(1, user))
+		if(O.density == 1)
+			if(O == user)
+				continue
+			if(O == target)
+				continue
+			if(O.opacity)
+				continue
+			else
+				surface = O
+				ST = get_turf(O)
+				break
+
+	if(surface && (ST && isturf(ST)))
+		user.forceMove(ST)
+		user.visible_message("<span class='danger'>[user] climbs onto [surface]!</span>", \
+						"<span class='danger'>You climb onto [surface]!</span>")
+		user.pixel_y = user.base_pixel_y + 10
+		falling = 1
+		sleep(10)
+	if(user && target)
+		if((falling == 0 && get_dist(user, target) > 1) || (falling == 1 && get_dist(user, target) > 2)) // We climbed onto stuff.
+			user.pixel_y = user.base_pixel_y
+			if(falling == 1)
+				user.visible_message("<span class='danger'>...and dives head-first into the ground, ouch!</span>", \
+								"<span class='userdanger'>...and dive head-first into the ground, ouch!</span>")
+				user.adjustBruteLoss(rand(10,15))
+				user.Paralyze(40)
+			to_chat(user, "<span class='warning'>[target] is too far away!</span>")
+			return
+		if(!isturf(user.loc) || !isturf(target.loc))
+			user.pixel_y = user.base_pixel_y
+			to_chat(user, "<span class='warning'>You can't drop onto [target] from here!</span>")
+			return
+		if(user)
+			animate(user, transform = matrix(90, MATRIX_ROTATE), time = 1, loop = 0)
+		sleep(10)
+		if(user)
+			animate(user, transform = null, time = 1, loop = 0)
+
+		user.forceMove(target.loc)
```

Comment:
you can deadass "nothin' personnel... kid..." people with this if they teleport away or something during the 1 second sleep()